setup.py

```diff
@@ -1,12 +1,27 @@
+import re
+import subprocess
 from distutils.core import setup
 
 from setuptools import find_packages
 
 import server
 
+
+def get_version() -> str:
+    output = subprocess.run(
+        ["git", "describe", "--abbrev=0", "--tags"],
+        capture_output=True
+    ).stdout.decode().strip()
+
+    m = re.match(r"v?(\d+\.\d+\.\d+)", output)
```

Comment:
I would say that second and third are not alwyas used.. `r"v?(\d+(\.\d+\(.\d+)?)?)"` o shorter `r"v?(\d+\.?)?"`
Like, somebody can write `v1` and somebody `v1.1`, and somebody `v1.1.1`

Reply:
Hm. That's probably true. I would prefer to use semver but I guess I can't guarantee that other people will always adhere to that.

Reply:
BTW: I've got idea! Find all `\d+` in version, and don't care about version format :-), then do everything you want with it. 

```python3
import re
re.findall("\d+", version)
```

Reply:
Oh yea that's not bad.

```python
".".join(re.findall(r"\d+", version)) or "dev"
```